### PR TITLE
[WASI-NN] Enable ggml llama.cpp plugin on macOS

### LINKED ISSUE #2811: Error while building WasiNN Plugin
## Description
I'm using the latest wasmedge 0.13.4 version. I'm not able to build the WasiNN Plugin. I'm trying to build the MediaPipe-rs module and I require WasiNN.

### Current State
Screenshot 2023-09-15 at 7 36 07 PM

## Environment

- Hardware Architecture: (Mac M1)
- Operating system: (MacOS)

The following information is optional. Please provide them only if you have built from source.
- C++ Compiler version:
- CMake version:   3.27.2
- CMake flags: `-DCMAKE_BUILD_TYPE=Release -DWASMEDGE_PLUGIN_WASI_NN_BACKEND="TensorflowLite"`

Commands executed - 
* cmake -GNinja -Bbuild -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_PLUGIN_WASI_NN_BACKEND="TensorflowLite"
* cmake --build build

### PR DESCRIPTION
**Pull request by hydai**
Fixes #2811 #2797

### CODE REVIEW

**Comment by hydai** [comment]
I forgot to update the workflow files. Working on them.

**Comment by hydai** [comment]
No need to support the macOS intel model. Need to fix the build failure on the intel model by disabling the ggml plugin.

File: plugins/wasi_nn/thirdparty/ggml/CMakeLists.txt [hunk 1]
```diff
@@ -51,7 +51,7 @@ set(LLAMA_CUDA_PEER_MAX_BATCH_SIZE "128" CACHE STRING
 option(LLAMA_HIPBLAS                         "llama: use hipBLAS"                               OFF)
 option(LLAMA_CLBLAST                         "llama: use CLBlast"                               OFF)
 option(LLAMA_METAL                           "llama: use Metal"                                 ${LLAMA_METAL_DEFAULT})
-option(LLAMA_METAL_NDEBUG                    "llama: disable Metal debugging"                   OFF)
+option(LLAMA_METAL_NDEBUG                    "llama: disable Metal debugging"                   ON)
```
Comment (dm4) [inline plugins/wasi_nn/thirdparty/ggml/CMakeLists.txt:54]: Should we set this option in `plugins/wasi_nn/CMakeLists.txt` for better integration when updating ggml?

File: plugins/wasi_nn/wasinnenv.cpp [hunk 2]
```diff
@@ -4,6 +4,8 @@
 #include "wasinnenv.h"
 #include "wasinnmodule.h"
 
+#include <sstream>
```
Comment (dm4) [inline plugins/wasi_nn/wasinnenv.cpp:7]: Is this for debugging?

Comment (hydai) [reply plugins/wasi_nn/wasinnenv.cpp:7]: Nope, this is for fixing https://github.com/WasmEdge/WasmEdge/issues/2811

**Review by hydai — COMMENTED** [review]
Hold on merging this. We have to ship the `libWasmEdgePluginWASINN.so` with the `Metal` file... Not only ship the shared library...

File: plugins/wasi_nn/thirdparty/ggml/CMakeLists.txt (hunk 1 above)
Comment (hydai) [reply plugins/wasi_nn/thirdparty/ggml/CMakeLists.txt:54]: Sounds better.

**Comment by hydai** [comment]
Still need time to figure out how to package the metal with a shared library.

**Review by dm4 — APPROVED** [review]

**Pull request merged**